lib/rules/no-redeclare.js

```diff
@@ -65,8 +91,91 @@ module.exports = {
 				yield { type: "builtin" };
 			}
 
-			for (const id of variable.identifiers) {
-				yield { type: "syntax", node: id, loc: id.loc };
+			const identifiers = variable.identifiers
+				.map(id => ({
+					identifier: id,
+					parent: id.parent,
+				}))
+				// ignore function declarations because TS will treat them as an overload
+				.filter(({ parent }) => parent.type !== "TSDeclareFunction");
+
+			// Handle TypeScript declaration merging if enabled
+			if (ignoreDeclarationMerge && identifiers.length > 1) {
+				// namespace/module merging
+				if (
+					identifiers.every(
+						({ parent }) => parent.type === "TSModuleDeclaration",
+					)
+				) {
+					return;
+				}
```

Comment:
Doesn't https://github.com/eslint/eslint/pull/19563#discussion_r2021370869 apply here, too?

Reply:
It does, missed this.